File: frontend/types/analytics.ts
export interface VelocityDataPoint {
  date: string
  merged_count: number
  open_count: number
  closed_count: number
}

export interface HeatmapCell {
  date: string
  count: number
  level: 0 | 1 | 2 | 3 | 4
}

export interface ContributorStats {
  username: string
  avatar_url: string
  pr_count: number
  merged_count: number
  merge_rate: number
  avg_days_to_merge: number
  lines_added: number
  lines_removed: number
  contribution_score: number
}

File: backend/app/schemas/analytics.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class VelocityDataPoint(BaseModel):
    date: str
    merged_count: int
    open_count: int
    closed_count: int


class HeatmapCell(BaseModel):
    date: str
    count: int
    level: Literal[0, 1, 2, 3, 4]


class ContributorStats(BaseModel):
    username: str
    avatar_url: str
    pr_count: int
    merged_count: int
    merge_rate: float
    avg_days_to_merge: float
    lines_added: int
    lines_removed: int
    contribution_score: float
